### ISSUE TITLE
Multiple selection tags, deselect/remove all if specific option selected. 

### ISSUE THREAD

**Issue by abzal0** [shentao/vue-multiselect#1465]
Issue: 
if one specific option is selected i am setting multiselect prop to false so that no other option can be selected along with it.  If specific option is selected it must clear all previously selected options and set current to selected. 

If I select specific option first, it works. If i select multiple options and then select specific option to remove all, it doesn't removes them. 

### Reproduction Link
https://jsfiddle.net/abz01/a2p1zq4o/ 

### Steps to reproduce
Set props     :multiple="isMultiselect" to control if it is multiple selection or not   and   @select="onSelect" to change isMultiselect value and empty selected. 
On select if option value is  'Remove all but keep this option' set model this.value to empty [] (to clear selection) and set this.isMultiselect = false (to make only one option selected). 
### Expected behaviour
![image](https://user-images.githubusercontent.com/5952350/127644594-392a9cad-d4d9-421a-b1b7-b0afa184a98c.png)

On select of specific option 'Remove all but keep this option' should remove 'One' and 'Two' from being selected. 

### Actual behaviour
Before: 
![image](https://user-images.githubusercontent.com/5952350/127645017-23560fb5-9669-4bfa-ad59-d1e89cb65750.png)
 
After selecting  'Remove all but keep this option'  from the list: 

![image](https://user-images.githubusercontent.com/5952350/127644624-36affcd7-41a6-46cc-a743-9da796e57e64.png)

Both options are still selected, it just makes first option to show up instead of clearing it.  

TLDR; 
How to clear multiple selected options if specific option is selected? Emptying model isn't working.

**Comment by abzal0** [comment]
Found the solution. 

this.value = [] wasn't working because `@select` was used instead of `@input ` . Realized this when console logged selected option and noticed it shows up only previously selected ignoring currently selected with `@select`.   

Solution: 
```

<div id="app">
  <multiselect 
    class="multiselect"  
    open-direction="bottom" 
    v-model="value" 
    :options="options" 
    track-by="title" 
    label="title"   
    placeholder="Type to search"  
    :multiple="isMultiselect"    
 
    @input="onSelect"  >
  </multiselect>
 </div>
 
 new Vue({
	components: {
  	Multiselect: window.VueMultiselect.default
	},
	data: {
    value: [],
          isMultiselect: true  ,
  	options: [
    	{	id: '1', title:"One" },
      {	id: '2', title:"Two" },
      {	id: '3', title:"Three" },
            {	id: '4', title:"Remove all but keep this option" },
    ]
	},
   methods: {
 
        onSelect (value) {
         
   console.log(this.value );
 
      
      
       if(value[value.length-1].id == 4) // downside of @input is that it gets all values where as @select gets latest, therefore getting latest value in array 
        {
  
 //setting only one selected and clearing all previous
      this.value = [
 {	id: '4', title:"Remove all but keep this option" }
      ]; 
 
 
   this.isMultiselect = false; //making selection to be not multiselect
        
        }
        else {
   this.isMultiselect = true; //if anything else selected it will become multiselect again

        }   
        
        
      
 

    },
 
    }
}).$mount('#app')


```
